src/app/utils.ts

```diff
@@ -60,7 +61,8 @@ const makeCatchError = ({dispatch, getState}: MiddlewareAPI) => (e: Error) => {
     return;
   } else if (e instanceof ToastMesssageError) {
     Sentry.captureException(e);
-    dispatch(addToast(e.messageKey, e.meta));
+    const errorId = getMessageIdStack(getState())[0];
+    dispatch(addToast(e.messageKey, { ...e.meta, errorId }));
```

Comment:
this kind of decoupling is worrying me... we're assuming that captureException is synchronous and there is risk that we could get the wrong or duplicate error ids out of the state if the order is not guaranteed.

going back to the implementation you had before, could we try doing something like this?
```diff
diff --git a/src/helpers/Sentry.ts b/src/helpers/Sentry.ts
index f16875301..d574a8909 100644
--- a/src/helpers/Sentry.ts
+++ b/src/helpers/Sentry.ts
@@ -50,15 +50,19 @@ export default {
     return typeof(window) !== 'undefined' && config.SENTRY_ENABLED;
   },
 
-  captureException(error: any, level: Sentry.Severity = Severity.Error) {
+  captureException(error: any, level: Sentry.Severity = Severity.Error): string | undefined {
+    let eventId;
+
     if (this.isEnabled) {
       Sentry.withScope((scope) => {
         scope.setLevel(level);
-        Sentry.captureException(error);
+        eventId = Sentry.captureException(error);
       });
     } else if (!this.shouldCollectErrors) {
       console.error(error); // tslint:disable-line:no-console
     }
+
+    return eventId;
   },
 
   captureMessage(message: string, level: Sentry.Severity) {
```

Reply:
I've written a comment about this https://github.com/openstax/rex-web/pull/1046#issuecomment-769821295
I'll push a change so we can test if this is working and typescript is just not recognizing properly the type

Reply:
because its a let you have to tell it to be `string | undefined`